File: backend/src/scripts/bootstrapAdmin.ts
import mongoose from "mongoose";
import User from "../models/User.js";
import dotenv from "dotenv";

dotenv.config();

const MONGO_URI = process.env.MONGO_URI || "mongodb://127.0.0.1:27017/your-db-name";

const admins = [
    {
        name: "Mehdi",
        lastName: "Kiona",
        phoneNumber: "+27659546977",
        role: "admin",
        status: "approved",
    },
    // Add more admin objects here as needed
    // {
    //   name: "Second",
    //   lastName: "Admin",
    //   phoneNumber: "+27820000001",
    //   role: "admin",
    //   status: "approved",
    // },
];

async function createAdmins() {
    await mongoose.connect(MONGO_URI);
    for (const admin of admins) {
        const existing = await User.findOne({ phoneNumber: admin.phoneNumber });
        if (existing) {
            console.log(`Admin with phone ${admin.phoneNumber} already exists.`);
        } else {
            await User.create(admin);
            console.log(`Admin user ${admin.phoneNumber} created!`);
        }
    }
    process.exit();
}

createAdmins().catch((err) => {
    console.error("Error creating admins:", err);
    process.exit(1);
});

File: LetGO_to_Icc-server/src/accounts/management/commands/bootstrap_admin.py
from django.core.management.base import BaseCommand

from accounts.models import User

ADMINS = [
    {
        "name": "Mehdi",
        "last_name": "Kiona",
        "phone_number": "+27659546977",
        "role": "admin",
        "status": "approved",
    },
    # Add more admin dicts here as needed
]


class Command(BaseCommand):
    help = "Create initial admin users if they don't already exist"

    def handle(self, *args, **options):
        for admin_data in ADMINS:
            phone = admin_data["phone_number"]
            if User.objects.filter(phone_number=phone).exists():
                self.stdout.write(f"Admin with phone {phone} already exists.")
            else:
                User.objects.create_user(**admin_data)
                self.stdout.write(self.style.SUCCESS(f"Admin user {phone} created!"))
